lib/jormungandr/test/unit/Cardano/Wallet/Jormungandr/CompatibilitySpec.hs

```diff
@@ -0,0 +1,208 @@
+            goldenTest
+                [ "df9f08672a3a94778229b91daa981538883e1535d666dc10e63b438f44c63e3f"
+                ]
+                "ta1s00e7zr89gafgauz9xu3m25cz5ugs0s4xhtxdhqsuca58r6ycclr7v3je63"
+            goldenTest
+                [ "7bd5386c31ac31ba7076856500cf26f85d4695b80f183c7a53e3f28419d6bde1"
+                , "b24e70b0c2ceeb24cc9f28f386478c73aa71c05a95a0119bb91dd8e89c3592ae"
+                ]
+                "ta1s3aa2wrvxxkrrwnsw6zk2qx0ymu96354hq83s0r6203l9pqe6677r\
+                \vjwwzcv9nhtynxf728nserccua2w8q949dqzxdmj8wcazwrty4we4spcz"
+            goldenTest
+                [ "df9f08672a3a94778229b91daa981538883e1535d666dc10e63b438f44c63e3f"
+                , "402abff6065c847115ad22ff6b0d3a85fd69a6fcc32ed76aa8cadb305b0c51a7"
+                ]
+                "ta1sn0e7zr89gafgauz9xu3m25cz5ugs0s4xhtxdhqsuca58r6ycclr7\
+                \sp2hlmqvhyywy266ghldvxn4p0adxn0esew6a423jkmxpdsc5d8xw6gar"
+
+  where
+    decodeAddress' = decodeAddress (Proxy @Jormungandr)
+    encodeAddress' = encodeAddress (Proxy @Jormungandr)
+    firstByteS = B8.unpack (BS.pack [single])
+    firstByteG = B8.unpack (BS.pack [grouped])
+    negativeTest input msg = it ("decodeAddress failure: " <> msg) $
+        decodeAddress' input === Left (TextDecodingError msg)
+
+-- | Generate addresses from the given keys and compare the result with an
+-- expected output.
+goldenTest
+    :: [ByteString]
+    -> Text
+    -> SpecWith ()
+goldenTest pubkeys expected = it ("golden test: " <> T.unpack expected) $ do
+    case traverse (convertFromBase Base16) pubkeys of
+        Right [spending] -> do
+            let payload = BS.pack [single] <> spending
+            let addr = encodeAddress (Proxy @Jormungandr) (Address payload)
+            addr `shouldBe` expected
+        Right [spending, delegation] -> do
+            let payload = BS.pack [grouped] <> spending <> delegation
+            let addr = encodeAddress (Proxy @Jormungandr) (Address payload)
+            addr `shouldBe` expected
+        _ ->
+            expectationFailure "goldenTest: provided invalid inputs public keys"
+
+{-------------------------------------------------------------------------------
+                             Arbitrary Instances
+-------------------------------------------------------------------------------}
+
+instance Arbitrary (ShowFmt Address) where
+    arbitrary = ShowFmt <$> frequency
+        [ (10, genAddress)
+        , (1, genLegacyAddress (30, 100))
+        ]
+
+genAddress :: Gen Address
+genAddress = oneof
+    [ (\bytes -> Address (BS.pack (single:bytes))) <$> vectorOf 32 arbitrary
+    , (\bytes -> Address (BS.pack (grouped:bytes))) <$> vectorOf 64 arbitrary
+    ]
+
+genLegacyAddress :: (Int, Int) -> Gen Address
+genLegacyAddress range = do
+    n <- choose range
+    let prefix = BS.pack
+            [ 130       -- Array(2)
+            , 216, 24   -- Tag 24
+            , 88, fromIntegral n -- Bytes(n), n > 23 && n < 256
```

Comment:
what if `n` is choosen out of the range "n > 23 && n < 256" ?

Reply:
This will result in a non-valid CBOR encoding. `88` is a major type which indicates that the next item is a bytestring and the length of that bytestring is encoded in the next byte `n`. So `n` can't be greater than 255 (it's a byte), and, in cbor, values lesser than `23` are encoded directly in the payload of the major type and thus, there would be no `n`. 

This is how most addresses are represented in Byron, although the address payload is malleable so it could be bigger than 255 bytes in practice; but it doesn't really matter much for the test here.